src/runtime/hexagon_remote/halide_hexagon_remote.idl

```diff
@@ -4,11 +4,12 @@ interface halide_hexagon_remote
     typedef unsigned long handle_t;
 
     // Routine to initialize a module on the remote side. Returns a pointer to the module.
-    long initialize_kernels(in buffer code, rout handle_t module_ptr);
+    long initialize_kernels_v2(in buffer code, in long use_shared_object,
+                               rout handle_t module_ptr);
 
     // Get a symbol from the module.
     // v2: return the result as an output parameter instead of a return value.
-    long get_symbol_v2(in handle_t module_ptr, in sequence<char> name, rout handle_t sym_ptr);
+    long get_symbol_v2(in handle_t module_ptr, in sequence<char> name, in long use_shared_object, rout handle_t sym_ptr);
```

Comment:
BTW, I can see why this was tricky to get right.

I think that the function in the IDL should be named v3.

Then, I think that there should be two shims in halide_host.cpp, get_symbol and get_symbol_v2, both of which call get_symbol_v3 (either directly or indirectly).

I think that maybe you are optimizing for consistency between the versions of the functions. I think that this is a good thing to do in some ways, but, we should also just optimize for the simplest/minimal thing in the latest version, because we will hopefully just remove some of the older versioning shims as soon as we can break this versioning stickyness. At that point, all we'd be left with is the newest version, so consistency with the old functions won't matter anymore.